docs/proposals/app-metrics.md

```diff
@@ -0,0 +1,124 @@
+ 
+## Requirements
+* support different application ports/paths
+* support more than one container/application
+* support no mTLS communication with Prometheus
+ 
+## Idea
+ 
+The idea is to create an aggregator in `kuma-dp`, to be specific Metrics Hijacker, that will be responsible for the aggregation of metrics from applications defined to scrap(within local env: pod/vm). Communication with `kuma-dp` might be configured as secure but it won't be required(and not recommended). It's worth mentioning that this idea can support only Prometheus endpoint scrapping and exposing metrics as Prometheus.
+ 
+### Universal
+At the mesh level, it will be possible to define all applications endpoints that should be scrapped by `kuma-dp` within a Mesh.
+```
+type: Mesh
+name: default
+metrics:
+ enabledBackend: prometheus-1
+ backends:
+ - name: prometheus-1
+   type: prometheus
+   conf:
+     skipMTLS: true
+     mtls:
+       require: true
+       tags: # tags that can be referred in Traffic Permission when metrics are secured by mTLS 
+         kuma.io/service: dataplane-metrics
+     aggregate:
+     - name: app
+       port: 1234
+       path: /metrics # defaults to /metrics
+     - name: opa
+       port: 8888
+       path: /metrics # defaults to /metrics
+```
+`Dataplane` configuration enables at which port/path `kuma-dp` is going to expose aggregated metrics from Envoy and other applications running within one workflow. Also, it's possible to override the `Mesh` configuration of endpoints to scrap by the `Dataplane` configuration. They are going to be identified by `name` and if `Mesh` and `Dataplane` configurations have definitions with the same name, then the `Dataplane` configuration has precedence before `Mesh` and is merged.
+```
+type: Dataplane
+mesh: default
+name: redis-1
+networking:
+ address: 192.168.0.1
+ inbound:
+ - port: 9000
+   servicePort: 1234
+   tags:
+     kuma.io/service: backend
+ metrics:
+   type: prometheus
+   conf:
+     skipMTLS: true
+     port: 1234
+     path: /non-standard-path
+     aggregate:
+     - name: app # overrides app from Mesh
+       port: 1236
+       path: /metrics # defaults to /metrics
+     # inherits opa from Mesh
+```
+ 
+### Kubernetes
+ 
+The same `Mesh` configuration works for K8s and Universal.
+```
+apiVersion: kuma.io/v1alpha1
+kind: Mesh
+metadata:
+ name: default
+spec:
+ metrics:
+   enabledBackend: prometheus-1
+   backends:
+   - name: prometheus-1
+     type: prometheus
+     conf:
+       skipMTLS: false
+       port: 5670
+       path: /metrics
+       tags: # tags that can be referred in Traffic Permission when metrics are secured by mTLS 
+         kuma.io/service: dataplane-metrics
+       aggregate:
+       - name: app
+         port: 1236
+         path: /metrics # defaults to /metrics
+       - name: opa
+         port: 8888
+         path: /metrics # defaults to /metrics
```

Comment:
To me exposing this in the mesh atm is not really a usable solution (your apps within a mesh are just too different).

I think we should keep it as dataplane options and introduce a policy later for this.

Reply:
Ok then :)